### PR TITLE
🐙 octavia-cli: package and install script

### LINKED ISSUE #10708: 🐙 octavia-cli: package into a docker image
The `octavia-cli` shall be packaged and released as a docker image.

## TODO 
* Make sure the docker image can be used as a CLI
* Provide an helper script / snippets to explain how to install
* Release the docker image on our docker hub repo

### PR DESCRIPTION
**Pull request by alafanechere**
## What
This PRs aims at improving the packaging of the CLI and easing its installation (https://github.com/airbytehq/airbyte/issues/10708)

## How
**Fixes**:
 * Add Jinja templates to the installed Octavia package: [octavia-cli/setup.py](https://github.com/airbytehq/airbyte/pull/10823/files#diff-bb1316ba95893223eda7869ab9ef19f59d9032609932fbcd820670adcf119696)
 * Add missing `pyyaml` dependency: [octavia-cli/setup.py](https://github.com/airbytehq/airbyte/pull/10823/files#diff-bb1316ba95893223eda7869ab9ef19f59d9032609932fbcd820670adcf119696)
 * Move Jinja `templates` directory from `./octavia_cli` to `./octavia_cli/generate/` + [octavia-cli/octavia_cli/generate/renderer.py](https://github.com/airbytehq/airbyte/pull/10823/files#diff-ca4e081b452f89797f291d7ea993565db6bb84a3a17c5aa8003c71d880bd2792)
 * Typo in a command name (missing plural): [octavia-cli/octavia_cli/list/commands.py](https://github.com/airbytehq/airbyte/pull/10823/files#diff-b7af152b7a08a6ee477d4226de36b827a4b38398a9bf3024e4efebb8e9bbf60f)

**Addition**:
* [octavia-cli/publish.sh](https://github.com/airbytehq/airbyte/pull/10823/files#diff-3a195daaed4f005150ee759f749be658819a4414327b45c6e3406ba6ff6eb909) to tag and publish a new version to Airbyte repository
* [octavia-cli/install.sh](https://github.com/airbytehq/airbyte/pull/10823/files#diff-8546a8b1f79f192a3374c13f65623849524b5ab297aa9d82acd82ea5f2ad9f99) to add `octavia` alias to user's shell profile. `octavia` is an alias to a `docker run` command
* [octavia-cli/README.md](https://github.com/airbytehq/airbyte/pull/10823/files#diff-2aec83a3abb4fe83085351ad3bbceece28c9ccde1d5d3c1264abaa14c2fe1eee): Update install instructions

## 🚨 User Impact 🚨
`octavia` can be installed

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@alafanechere minor one can you update the doc tests?
```
6. Run the test suite: `pytest --cov=octavia_cli unit_tests`
```
to run with only tests folder? :D

File: octavia-cli/install.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    else
+        install
+    fi
+}
+
+set -e
+check_docker_is_running
+detect_profile
+set -u
+update_or_install
```
Comment (marcosmarxm) [inline octavia-cli/install.sh:73]: Great! 
image

File: octavia-cli/README.md [hunk 2]
```diff
@@ -10,23 +10,23 @@ It has the following features:
...
+## 2.b If you want to directly run the CLI without alias:
 ```bash
-docker run airbyte/octavia-cli:dev 
-````
-3. Create an `octavia` alias in your `.bashrc` or `.zshrc`: 
-````bash
-echo 'alias octavia="docker run airbyte/octavia-cli:dev"'  >> ~/.zshrc
-source ~/.zshrc
-octavia
+docker run --rm -v {}:/home/octavia-project --network host -e AIRBYTE_URL="${AIRBYTE_URL}" -e AIRBYTE_WORKSPACE_ID="${AIRBYTE_WORKSPACE_ID}" airbyte/octavia-cli:dev
```
Comment (marcosmarxm) [inline octavia-cli/README.md:29]: @alafanechere how do you will define the local path to map the folder and store the files?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Only one comment about the path to local instance when using docker.
Overall the script is working a

File: octavia-cli/README.md [hunk 3]
```diff
@@ -10,23 +10,23 @@ It has the following features:
...
+## 2.b If you want to directly run the CLI without alias in your current directory:
 ```bash
-docker run airbyte/octavia-cli:dev 
-````
-3. Create an `octavia` alias in your `.bashrc` or `.zshrc`: 
-````bash
-echo 'alias octavia="docker run airbyte/octavia-cli:dev"'  >> ~/.zshrc
-source ~/.zshrc
-octavia
+docker run --rm -v ${PWD}:/home/octavia-project --network host -e AIRBYTE_URL="${AIRBYTE_URL}" -e AIRBYTE_WORKSPACE_ID="${AIRBYTE_WORKSPACE_ID}" airbyte/octavia-cli:dev
```
Comment (alafanechere) [inline octavia-cli/README.md:29]:
```suggestion
docker run --rm -v ${PWD}:/home/octavia-project --network host -e AIRBYTE_URL="${AIRBYTE_URL}" airbyte/octavia-cli:dev
```

**Review by marcosmarxm — APPROVED** [review]
LGTM

**Pull request merged**